### ISSUE TITLE
Issue with Recent Updates & Virtual File System

### ISSUE THREAD

**Issue by ThePapaw** [RPCS3/rpcs3#13127]
### Quick summary

Recent Updates when using Virtual File System Taking close to 10 minutes to do anything after loading even to change a setting

### Details

I have 1.65TB of PS3 game data installed on a shared NAS drive that is mapped to PC via SAMBA

I was running "RPCS3 v0.0.25-14422-9274b3f1 Alpha | master" using Virtual File System pointing to the drive and all works as expected.

I updated to "RPCS3 v0.0.25-14495-8ac99680 Alpha | master" and it now takes around 8 minutes to go into anything, such as settings, launch a game or anything. It is scanning the entire virtual file system before doing anything (take 8 minutes or so). If I go into settings and wait the 8 minutes, then try to click another menu item, another 8 minute wait.
If I try to close RPCS3 before it scans the entire data system, I get the 12 errors in the image.

UPDATE: I waited a couple days and tried to update to version rpcs3-v0.0.25-14502-6c742745. It does the same thing for me

I posted in the forums a few days back, but I haven't gotten back any replies. I hope it was okay to post it here. Thanks for all the continuous great work

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/10303121/RPCS3.log)

### Attach capture files for visual issues

![Untitled](https://user-images.githubusercontent.com/51170894/209541373-286c572d-2398-40b1-837c-0c0ff636cd58.jpg)

**Comment by ThePapaw** [comment]
I forgot to include specs:

Win 10 (up-to-date)
Intel 8086k CPU
1080ti (drivers up-to-date)
32gb ram

**Comment by readywer** [comment]
You shoud test which build broke it, beacuse there are more than 30 build between 14422-14495.

**Comment by solarmystic** [comment]
@ThePapaw try https://github.com/RPCS3/rpcs3-binaries-win/releases/download/build-f54da22f6f4146ffb8b2bbe93559e5b789cbba7c/rpcs3-v0.0.25-14480-f54da22f_win64.7z 

Does that build resolve your issue?

**Comment by ThePapaw** [comment]
@solarmystic: Build 14479 appears to be working as expected and build 14480 does introduce the issue but not exactly the same.

With 14480 it takes minutes to open RPCS3 (not like 14479 or previous versions) but If I click on the config button it does open as expected and doesn't take minutes like with current version, however, if I type anything into the search box, it DOES takes minutes.

Log of 14480
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10307659/RPCS3.log.gz)

Any help I can be, please let me know.

@readywer thx for replying, will do next time, sorry I hadn't thought about that

**Comment by Udderdude** [comment]
This is probably the culprit. https://github.com/RPCS3/rpcs3/pull/13088

It's prob. taking forever to calculate the space of all games over a NAS. It's also pretty slow on an internal HDD or USB attached HDD.

**Comment by solarmystic** [comment]
@ThePapaw Try this test build based on #13134 

https://cirrus-ci.com/task/5293244129476608

Download it from the Artifacts section further down the page and run it. 

![image](https://user-images.githubusercontent.com/4345150/209753383-19776dcf-6d7c-44ae-9c59-539549cc6780.png)

Does it help out?

**Comment by ThePapaw** [comment]
@solarmystic It launches as normal, but still having 8 minute issue to try to change anything by hitting the config button as well as things like the search bar.

After the initial scan though, it does appear to be working right and the functions aren't taking minutes. Until I close RPCS3 and re-open it. Much improved, sorta . . . 

Attached log
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/10315719/RPCS3.log.gz)

**Comment by ThePapaw** [comment]
Has there been any progress on this?

**Comment by Behodar** [comment]
I'm having what appears to be the same issue. I'm on a Mac, but my games are on an SMB-attached NAS and with build 14483 or later I get the "spinning beachball" for about 30 seconds whenever I try to do anything. The previous Mac build, 14479, is fine.

- SYS: RPCS3 v0.0.25-14483-8ec01450 Alpha | master
- SYS: Intel(R) Core(TM) i9-10910 CPU @ 3.60GHz | 20 Threads | 16.00 GiB RAM | TSC: 3.600GHz | AVX+ | FMA3
- SYS: Operating system: macOS, Version: 12.6.4
- SYS: Current Time: 2023-04- 1T14:23:19
- RSX: Found vulkan-compatible GPU: 'AMD Radeon Pro 5700 XT' running on driver 0.2.894
- SYS: Firmware version: 4.88

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/11128154/RPCS3.log)

**Comment by ThePapaw** [comment]
I am still having the issue with current builds, taking minutes to get into settings, just not getting the errors as I was

**Comment by Udderdude** [comment]
Yes, it's still extremely slow and appears to have frozen up while it's sitting there calculating file space, especially if you have more than a handful of games installed. I would suggest adding an option to disable this feature.

**Comment by elad335** [comment]
Test with #13715

**Comment by DocMAX** [comment]
starting games from NAS (samba share) just freezes RPCS3. is this the problem here?

**Comment by Behodar** [comment]
If build 14479 works and 14480 freezes, then yes, that's the same issue.

**Comment by KingKrouch** [comment]
Any progress? Been using builds on Linux, and for some reason I cannot for the life of me add a virtual file system through the settings.

[RPCS3 Logs.zip](https://github.com/user-attachments/files/16738833/RPCS3.Logs.zip)

[RPCS3.log](https://github.com/user-attachments/files/16738834/RPCS3.log)

**Comment by digant73** [comment]
@ThePapaw is that still an issue?

**Comment by ThePapaw** [comment]
@digant73 - No, this was fixed for me a very long time ago
